inference-engine/src/inference_engine/src/cpp/ie_executable_network.cpp

```diff
@@ -22,6 +23,14 @@ namespace InferenceEngine {
         InferenceEngine::details::Rethrow();                                \
     }
 
+#define OV_EXEC_NET_CALL_STATEMENT(...)                                          \
+    OPENVINO_ASSERT(_impl != nullptr, "ExecutableNetwork was not initialized."); \
+    try {                                                                        \
+        __VA_ARGS__;                                                             \
+    } catch (const std::exception& ex) {                                         \
+        throw ov::Exception(ex.what());                                          \
```

Comment:
What if an exception that was not inherited from std::exception is thrown. I think it is better to introduce an ov::rethrow() helper that can handle such cases too. 
BTW: have a look to https://en.cppreference.com/w/cpp/error/throw_with_nested may be we can use it?

Reply:
Added `...` exceptions support